packages/mdc-radio/index.ts

```diff
@@ -22,106 +22,97 @@
+  set checked(checked: boolean) {
     this.nativeControl_.checked = checked;
   }
 
-  /** @return {boolean} */
   get disabled() {
     return this.nativeControl_.disabled;
   }
 
-  /** @param {boolean} disabled */
-  set disabled(disabled) {
+  set disabled(disabled: boolean) {
     this.foundation_.setDisabled(disabled);
   }
 
-  /** @return {?string} */
   get value() {
     return this.nativeControl_.value;
   }
 
-  /** @param {?string} value */
-  set value(value) {
+  set value(value: string) {
     this.nativeControl_.value = value;
   }
 
-  /** @return {!MDCRipple} */
-  get ripple() {
+  get ripple(): MDCRipple {
     return this.ripple_;
   }
 
-  constructor(...args) {
-    super(...args);
-
-    /** @private {!MDCRipple} */
+  // tslint:disable-next-line:no-any a component can pass in anything it needs to the constructor
+  constructor(root: Element, ...args: any[]) {
+    super(root, ...args);
     this.ripple_ = this.initRipple_();
   }
 
-  /**
-   * @return {!MDCRipple}
-   * @private
-   */
-  initRipple_() {
+  destroy() {
+    this.ripple_.destroy();
+    super.destroy();
+  }
+
+  getDefaultFoundation() {
+    return new MDCRadioFoundation({
+      addClass: (className) => this.root_.classList.add(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+      setNativeControlDisabled: (disabled) => this.nativeControl_.disabled = disabled,
+    });
+  }
+
+  private initRipple_(): MDCRipple {
     const adapter = Object.assign(MDCRipple.createAdapter(this), {
-      isUnbounded: () => true,
+      deregisterInteractionHandler:
+        <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
+          this.nativeControl_.removeEventListener(type, handler),
       // Radio buttons technically go "active" whenever there is *any* keyboard interaction. This is not the
       // UI we desire.
       isSurfaceActive: () => false,
-      registerInteractionHandler: (type, handler) => this.nativeControl_.addEventListener(type, handler),
-      deregisterInteractionHandler: (type, handler) => this.nativeControl_.removeEventListener(type, handler),
+      isUnbounded: () => true,
+      registerInteractionHandler: <K extends EventType>(type: K, handler: SpecificEventListener<K>) =>
+        this.nativeControl_.addEventListener(type, handler),
     });
     const foundation = new MDCRippleFoundation(adapter);
     return new MDCRipple(this.root_, foundation);
   }
 
   /**
    * Returns the state of the native control element, or null if the native control element is not present.
-   * @return {?MDCSelectionControlState}
-   * @private
    */
-  get nativeControl_() {
+  private get nativeControl_(): HTMLInputElement {
     const {NATIVE_CONTROL_SELECTOR} = MDCRadioFoundation.strings;
-    const el = /** @type {?MDCSelectionControlState} */ (
-      this.root_.querySelector(NATIVE_CONTROL_SELECTOR));
+    const el = this.root_.querySelector(NATIVE_CONTROL_SELECTOR) as HTMLInputElement;
```

Comment:
Instead of type casting, use the generic form: `querySelector<HTMLInputElement>()`.

Reply:
Done